.github/workflows/release-tarball.yml

```diff
@@ -0,0 +1,35 @@
+name: Release Tarball
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  upload-release-tarball:
+    name: upload complete source code for release
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: checkout tagged release recursively incl. submodules
+      uses: actions/checkout@v3
+      with:
+        submodules: recursive
+
+    - name: set environment variables
+      run: |
+        NAME="${{ github.repository }}"
+        NAME="${NAME#*/}-full-${{ github.ref_name }}"
+        FILE="./${NAME}.tar.gz"
```

Comment:
Again, the naming is generic. A more descriptive name like `RELEASE_TARBALL_PATH` would offer more clarity and avoid variable name collisions.